base/loading.jl

```diff
@@ -1044,7 +1044,17 @@ const TIMING_IMPORTS = Threads.Atomic{Int}(0)
 # these return either the array of modules loaded from the path / content given
 # or an Exception that describes why it couldn't be loaded
 # and it reconnects the Base.Docs.META
-function _include_from_serialized(pkg::PkgId, path::String, ocachepath::Union{Nothing, String}, depmods::Vector{Any})
+function _include_from_serialized(pkg::PkgId, path::String, ocachepath::Union{Nothing, String}, depmods::Vector{Any}, ignore_native::Union{Nothing,Bool}=nothing)
+    if isnothing(ignore_native)
+        io = open(path, "r")
+        try
+            iszero(isvalid_cache_header(io)) && return ArgumentError("Invalid header in cache file $path.")
```

Comment:
Are we now parsing the cache header twice? Once here an another time in the caller?

Reply:
`_include_from_serialized` is called by three functions. One that parses the header and that passes a Bool to `ignore_native`. The other two don't seem to parse the header so don't set `ignore_native` and leave it to default of `nothing` that triggers this header parse

Reply:
Added a fast path if coverage isn't on that avoids the header parse